dpcpp/components/atomic.dp.hpp

```diff
@@ -114,6 +126,19 @@ struct atomic_helper {
 };
 
 
+template <cl::sycl::access::address_space addressSpace, typename ValueType,
+          typename = void>
+struct atomic_max_helper {
+    __dpct_inline__ static ValueType atomic_max(ValueType *, ValueType)
+    {
+        static_assert(sizeof(ValueType) == 0,
+                      "This default function is not implemented, only the "
+                      "specializations are.");
+        // TODO: add proper implementation of generic atomic max
```

Comment:
We have atomic_max_helper, is that not sufficiently generic?

Reply:
It is copy-paste from cuda